### ISSUE TITLE
Please provide debian/ubuntu PPAs

### ISSUE THREAD

**Issue by teunvink** [fujiapple852/trippy#859]
Thanks for your nice tool! I manage a set of machines (https://ring.nlnog.net) on which this tool would be a nice addition. However, these machines do not run snap or docker, so that only leaves me with installing binaries directly on these machines. And that's something we prefer not to do, because it's harder to keep track of binaries and especially to update them when they are updated upstream.

I see that you build `.deb` files and link them on the project page, but these are only available on a changing URL containing the version number. Would it be possible to provide a PPA  for trippy?

If that is not possible, a stable link (not containing any version numbers) to the latest deb could be useful as well. Now, we'd have to parse the table on https://trippy.cli.rs/#binary-asset-download to determine the URL of the latest version of trippy so we can package it (or copy the binary if we wouldn't package it) if it changed.

**Comment by c-git** [comment]
Hi @teunvink,

Thanks for your request. I will look into what would be required to have a stable link setup for you. 

Regarding the PPA Option, are you familiar with the PPA setup process / able to help with getting that setup?

I'd also point out (although I suspect you may already be aware but just in case) that if you have cargo on these computers you can also use cargo install to update.

**Comment by fujiapple852** [comment]
Hi there @teunvink,

I must confess I wasn't aware of PPA before, I had though that the recommended way of packing software for Debian/Ubuntu was Snap and Deb files?  Though I am far from an expert on that!  But in any case, I would like for Trippy to be available in as many places as possible and do not object to adding a PPA.

I spent a little time reading up on it and eventually found [this](https://blog.zhimingwang.org/packaging-rust-project-for-ubuntu-ppa) guide for Rust and it seems, complicated.  Would you be able to guide us through the process of creating this for Trippy?  if not do you have any contacts who could?

I created a [placeholder](https://launchpad.net/~fujiapple/+archive/ubuntu/trippy) for Trippy in launchpad.net for now.

Ideally this PPA package would be owned and maintained by a package maintainer closer to the Debian/Ubuntu universe than me.  Better yet, what would be involved in making Trippy available as an official package in Debian/Ubuntu such that the PPA is not needed?

As for a stable link, I'm less keen on this as I wouldn't want to commit to never again changing urls, directory layouts, filenames etc.  Also I do not own or control `cli.rs` (it is a service provided by [cli.rs](https://cli.rs/)) so I cannot make stability guarantees on their behalf.  

Trippy releases are infrequent, of the order of quarterly, so I'd hope updating some config in ring (which looks like a really great!) every now and again wouldn't be to much of a burden?

**Comment by teunvink** [comment]
You are correct that deb is the default package format for Debian and Ubuntu based setups. However, downloading and installing deb-files isn't the normal way to do it. Packages are either included in the default repository, or a PPA is provided as an endpoint where the developer publishes versions of the deb files.

I don't have any experience in hosting PPA's, but I'm happy to ask around if anyone is able to help out.

**Comment by teunvink** [comment]
I know next to nothing about rust either, but maybe this is useful? https://blog.zhimingwang.org/packaging-rust-project-for-ubuntu-ppa#impatient

**Comment by fujiapple852** [comment]
@teunvink it looks like you found the same guide I mentioned to above :)

Do let me know if you find someone who is able to own or assist with this and i'd be happy to work with them to get a PPA created.

**Comment by c-git** [comment]
@teunvink I was looking at another tool today and saw a way to get the latest version as part of their [install script](https://github.com/atuinsh/atuin/blob/a4122f062acdf7ccaf8b81acf35056b7c6f28c5e/install.sh#L33). I'm not sure something like that would be a temporary solution to the challenge that your facing. Let me know if it helps (mostly just curious but the PPA option would still be good if someone would want to take the lead on that).

**Comment by c-git** [comment]
I see another project setting up a PPA actually doesn't seem too bad other than key management. https://github.com/atuinsh/atuin/issues/1497

**Comment by teunvink** [comment]
> Do let me know if you find someone who is able to own or assist with this and i'd be happy to work with them to get a PPA created.

Sadly, I haven't really found someone yet. Packaging go stuff seems to be a bit unpopular (at least with the people I am in contact with).

I found a small script we use in our Ansible repository to make a deb-repository to host the PPA's for our customized tools, you can see it here: https://github.com/NLNOG/ring-ansible/blob/master/roles/apt-repository/files/deb2repo.

Since you already have .deb's, making a repo out of them seems to be a small step (other than hosting it perhaps).

**Comment by fujiapple852** [comment]
@teunvink Trippy now has a PPA: https://launchpad.net/~fujiapple/+archive/ubuntu/trippy

Currently only available for `Jammy` on `amd64`. 

To install:

```
sudo add-apt-repository ppa:fujiapple/trippy
sudo apt update && apt install trippy
```

All credit goes to @zarkdav.
